providers/aws/connection/connection_test.go

```diff
@@ -0,0 +1,42 @@
+package connection
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFirstNonEmptyRegion(t *testing.T) {
+	tests := []struct {
```

Comment:
this test is not much, but its a foundation to see if we can add more tests to the connection package going fwd.